packages/dates/src/utilities/formatDate.ts

```diff
@@ -1,9 +1,13 @@
-import {memoize} from '@shopify/function-enhancers';
-
-const memoizedGetDateTimeFormat = memoize(
-  getDateTimeFormat,
-  dateTimeFormatCacheKey,
-);
+const intl = new Map();
+const memoizedGetDateTimeFormat = function(locale, options) {
```

Comment:
Why does this not cause leaks but the implementation from `function-enhancers ` does? Could we fix it there so that we don't risk this happening again in the future?

Reply:
The function enhancers implementation doesn't memoize when server side rendering. https://github.com/Shopify/quilt/blob/master/packages/function-enhancers/src/memoize.ts#L18

Reply:
I don't know what the reasoning was behind not memoizing when SSRing, I mostly just wanted to try this out if that fixes things. It does locally, but wanted to see it in production. Happy to change this later on.

Reply:
SSR means the react app only render the one time. 
This mean memoize on server actually does not enhance performance, but leave behind memozing that will never be clear off.

this is the reasoning in theory, but maybe in practice there are some thing we are missing?

Reply:
In the past we've had memory leaks from memoizing on the server, so that was the idea between not doing so. I fear this might make things worse instead of better.

Reply:
That said, if it does work locally I suppose it's worth a shot.

Reply:
Explanation for not memoizing on the server makes sense. 👍 

> It worries me that we have no real rationale for why it would work though :(

I think the rationale for this is that there is a v8/Node.js issue which leads to `Intl.DateTimeFormat` not being properly cleaned up by the GC, and since we don't memoize we create more and more of those objects. So memoizing those objects minimizes the overall amount of objects we'll create which ends up in less unreachable objects for the GC.

Reply:
if we have legit reasons to not skip over server memorization (which I assume could happen else where), then lets add an option to `memoize` to specifically memorize no matter what.

Reply:
I'll try what I have now in production, if that works I'm happy to change the memoizer. 👍